sycl/include/CL/sycl/handler.hpp

```diff
@@ -426,6 +426,13 @@ class handler {
     KernelFunc(detail::Builder::getItem<Dims, true>());
   }
 
+  template <typename KernelName, typename KernelType, int Dims,
+            EnableIfNDItem<KernelType, Dims> = 0>
+  __attribute__((sycl_kernel)) void
+  kernel_parallel_for_nd_range(KernelType KernelFunc) {
+    KernelFunc(detail::Builder::getNDItem<Dims>());
+  }
+
   template <typename KernelName, typename KernelType, int Dims,
```

Comment:
Shouldn't we remove this function then?

Reply:
> Shouldn't we remove this function then?

I agree with you since only nd_item can only work along with nd_range

Reply:
Sorry, I found a testcase where range and nd_item are used by parallel_for. In this case Localsize is implicitly set to 1, and it seems we have a underlying nd_range. 
PLS: https://github.com/intel/llvm/blob/sycl/sycl/test/basic_tests/parallel_for_range_host.cpp.
So, I think we should not delete this function
```
// parallel_for, 30 global, 1(implicit) local -> pass.
  try {
    Q.submit([&](handler &CGH) {
        CGH.parallel_for<class g>(range<1>(30),
            [=](nd_item<1>) {});
    });
    Q.wait_and_throw();
  } catch (nd_range_error) {
    std::cerr << "Test case 'g' failed: exception has been thrown" << std::endl;
    return 1;
  }
```

Reply:
I seems it is an error. I don't see the SYCL specification allows range + nd_item combination.

Reply:
> I seems it is an error. I don't see the SYCL specification allows range + nd_item combination.

Yeah, I agree with you. Besides, I've fixed it in this PR, as well as the wrong testcase.